test/benchmarks/src/main/scala/scala/tools/nsc/transform/patmat/ClassMatchBenchmark.scala

```diff
@@ -0,0 +1,1127 @@
+        case 125 => "125" case 126 => "126" case 127 => "127" case 128 => "128" case 129 => "129"
+        case 130 => "130" case 131 => "131" case 132 => "132" case 133 => "133" case 134 => "134"
+        case 135 => "135" case 136 => "136" case 137 => "137" case 138 => "138" case 139 => "139"
+        case 140 => "140" case 141 => "141" case 142 => "142" case 143 => "143" case 144 => "144"
+        case 145 => "145" case 146 => "146" case 147 => "147" case 148 => "148" case 149 => "149"
+        case 150 => "150" case 151 => "151" case 152 => "152" case 153 => "153" case 154 => "154"
+        case 155 => "155" case 156 => "156" case 157 => "157" case 158 => "158" case 159 => "159"
+        case 160 => "160" case 161 => "161" case 162 => "162" case 163 => "163" case 164 => "164"
+        case 165 => "165" case 166 => "166" case 167 => "167" case 168 => "168" case 169 => "169"
+        case 170 => "170" case 171 => "171" case 172 => "172" case 173 => "173" case 174 => "174"
+        case 175 => "175" case 176 => "176" case 177 => "177" case 178 => "178" case 179 => "179"
+        case 180 => "180" case 181 => "181" case 182 => "182" case 183 => "183" case 184 => "184"
+        case 185 => "185" case 186 => "186" case 187 => "187" case 188 => "188" case 189 => "189"
+        case 190 => "190" case 191 => "191" case 192 => "192" case 193 => "193" case 194 => "194"
+        case 195 => "195" case 196 => "196" case 197 => "197" case 198 => "198" case 199 => "199"
+        case 200 => "200" case 201 => "201" case 202 => "202" case 203 => "203" case 204 => "204"
+        case 205 => "205" case 206 => "206" case 207 => "207" case 208 => "208" case 209 => "209"
+        case 210 => "210" case 211 => "211" case 212 => "212" case 213 => "213" case 214 => "214"
+        case 215 => "215" case 216 => "216" case 217 => "217" case 218 => "218" case 219 => "219"
+        case 220 => "220" case 221 => "221" case 222 => "222" case 223 => "223" case 224 => "224"
+        case 225 => "225" case 226 => "226" case 227 => "227" case 228 => "228" case 229 => "229"
+        case 230 => "230" case 231 => "231" case 232 => "232" case 233 => "233" case 234 => "234"
+        case 235 => "235" case 236 => "236" case 237 => "237" case 238 => "238" case 239 => "239"
+        case 240 => "240" case 241 => "241" case 242 => "242" case 243 => "243" case 244 => "244"
+        case 245 => "245" case 246 => "246" case 247 => "247" case 248 => "248" case 249 => "249"
+        case 250 => "250" case 251 => "251" case 252 => "252" case 253 => "253" case 254 => "254"
+        case 255 => "255"
+      }
+      bh.consume(x)
+      i += 1
+    }
+  }
+
+  @Benchmark @OperationsPerInvocation(count) def classNameHashSwitchShow(bh: Blackhole): Unit = {
+    val names = this.names
+    var i = 0
+    while (i < names.length) {
+      val name = names(i)
+      val cls  = name.getClass
+      val x = ((cls.getName.##): @switch) match {
```

Comment:
Is the implementation of `hashCode` for String considered to be stabled (between jdks)?

Reply:
Seems so https://stackoverflow.com/a/15693588/463761